src/lib/completion.ts

```diff
@@ -1,131 +1,216 @@
 
 const DEFAULT_MODEL = 'gpt-3.5-turbo';
 const DEFAULT_COLLECTION = 'default';
-
 export class ChatbotCompletion {
-	private model: ChatOpenAI;
 	private embeddings_model: OpenAIEmbeddings;
-
-	private qdrant_client: QdrantClient;
-	private qdrant_collection: string;
+	private executor: AgentExecutor | undefined;
+	private openai_api_key: string;
+	private model_name;
+	private verbose: boolean;
+	private qdrantClient: QdrantClient;
+	private collection_name: string;
+	private history: ChatHistory[];
+	private do_history: boolean;
+	private generate_data: boolean;
+	private chain: string[];
+	private summaryBot: OpenAI;
+	private do_summaryBot: boolean;
 
 	constructor(
 		openai_api_key: string,
 		{
 			openai_model = DEFAULT_MODEL,
-			qdrant_collection = DEFAULT_COLLECTION
+			collection_name = DEFAULT_COLLECTION,
+			verbose = false,
+			do_history = true,
+			generate_data = false,
+			do_summaryBot = true
 		}: {
 			openai_model?: string;
-			qdrant_collection?: string;
+			collection_name?: string;
+			verbose?: boolean;
+			do_history?: boolean;
+			generate_data?: boolean;
+			do_summaryBot?: boolean;
 		}
 	) {
-		this.model = new ChatOpenAI({
-			openAIApiKey: openai_api_key,
-			temperature: 0.7,
-			streaming: true,
-			maxTokens: 250,
-			modelName: openai_model,
-			verbose: true
-		});
+		this.verbose = verbose;
+		this.model_name = openai_model;
+		this.openai_api_key = openai_api_key;
 
 		this.embeddings_model = new OpenAIEmbeddings({
 			openAIApiKey: openai_api_key,
 			modelName: 'text-embedding-ada-002'
 		});
-		this.qdrant_client = new QdrantClient({
-			url: 'http://' + (process.env.QDRANT_HOST ?? 'localhost') + ':6333'
+		this.summaryBot = new OpenAI({
+			openAIApiKey: openai_api_key,
+			modelName: this.model_name,
+			temperature: 0.0
 		});
-
-		this.qdrant_collection = qdrant_collection;
+		this.do_summaryBot = do_summaryBot;
+		this.qdrantClient = new QdrantClient({ host: 'localhost', port: 6333 });
+		this.collection_name = collection_name;
+		this.executor = undefined;
+		this.history = [];
+		this.do_history = do_history;
+		this.generate_data = generate_data;
+		this.chain = [];
+		if (openai_api_key == undefined) {
+			throw console.warn('OPENAI_API_KEY is undefined');
+		}
 	}
 
-	/*
-        We need to pass with_vector to qdrant to get our response
-    */
-	private async qdrant_similarity_search(query: string, k: number): Promise<Document[]> {
-		const query_embedding = await this.embeddings_model.embedQuery(query);
-		const qdrant_results = await this.qdrant_client.search(this.qdrant_collection, {
-			vector: query_embedding,
-			limit: k,
-			with_vector: true
+	public async setup() {
+		console.log(colors.fg.cyan, 'Model name: ' + this.model_name, colors.style.reset);
+		console.log('\n');
+		const tools = getTools(
+			this.qdrantClient,
+			this.collection_name,
+			this.embeddings_model,
+			this.summaryBot,
+			this.do_summaryBot
+		);
+		const model = new ChatOpenAI({
```

Comment:
creating the chatbot isn't async and shouldn't need to be called in this function. We can do it in the constructor.